R/rdeephaven/tests/testthat/test_table_handle_wrapper.R

```diff
@@ -0,0 +1,173 @@
+                "th1" = th1, "th2" = th2, "th3" = th3, "th4" = th4))
+}
+
+##### TESTING GOOD INPUTS #####
+
+test_that("is_static returns the correct value", {
+    data <- setup()
+
+    expect_true(data$th1$is_static())
+    expect_true(data$th2$is_static())
+    expect_true(data$th3$is_static())
+    expect_true(data$th4$is_static())
+
+    # TODO: test ticking tables when they can be created from R
+})
+
+test_that("nrow returns the correct number of rows", {
+    data <- setup()
+
+    expect_equal(data$th1$nrow(), nrow(data$df1))
+    expect_equal(data$th2$nrow(), nrow(data$df2))
+    expect_equal(data$th3$nrow(), nrow(data$df3))
+    expect_equal(data$th4$nrow(), nrow(data$df4))
+
+    # TODO: test nrow(data$th) when it is implemented
+})
+
+test_that("bind_to_variable binds the table to a variable", {
+    data <- setup()
+
+    data$th1$bind_to_variable("table1")
+    expect_no_error(data$client$open_table("table1"))
+
+    data$th2$bind_to_variable("table2")
+    expect_no_error(data$client$open_table("table2"))
+
+    data$th3$bind_to_variable("table3")
+    expect_no_error(data$client$open_table("table3"))
+
+    data$th4$bind_to_variable("table4")
+    expect_no_error(data$client$open_table("table4"))
+})
+
+test_that("to_arrow_record_batch_stream_reader returns an identical stream reader", {
+    data <- setup()
+
+    # actual equality of RecordBatchStreamReaders is not expected, as they contain underlying pointers to relevant data,
+    # as well as metadata from the Deephaven server, which we do not expect rbr created fully in R to have.
+    # We care about equality in the sense that coercing to dataframes should yield identical dataframes, so we cast to dataframes and compare.
+
+    rbr1 <- data$th1$to_arrow_record_batch_stream_reader()
+    expect_equal(as.data.frame(as.data.frame(rbr1$read_table())), data$df1)
+    
+    rbr2 <- data$th2$to_arrow_record_batch_stream_reader()
+    expect_equal(as.data.frame(as.data.frame(rbr2$read_table())), data$df2)
+
+    rbr3 <- data$th3$to_arrow_record_batch_stream_reader()
+    expect_equal(as.data.frame(as.data.frame(rbr3$read_table())), data$df3)
+
+    rbr4 <- data$th4$to_arrow_record_batch_stream_reader()
+    expect_equal(as.data.frame(as.data.frame(rbr4$read_table())), data$df4)
+})
+
+test_that("to_arrow_table returns a valid Arrow table", {
+    data <- setup()
+
+    # The rationale for casting RecordBatchStreamReaders to dataframes for comparison also applies to Arrow Tables.
+
+    arrow_tbl1 <- data$th1$to_arrow_table()
+    expect_equal(as.data.frame(as.data.frame(arrow_tbl1)), data$df1)
+    
+    arrow_tbl2 <- data$th2$to_arrow_table()
+    expect_equal(as.data.frame(as.data.frame(arrow_tbl2)), data$df2)
+
+    arrow_tbl3 <- data$th3$to_arrow_table()
+    expect_equal(as.data.frame(as.data.frame(arrow_tbl3)), data$df3)
+
+    arrow_tbl4 <- data$th4$to_arrow_table()
+    expect_equal(as.data.frame(as.data.frame(arrow_tbl4)), data$df4)
```

Comment:
comment

Are the multiple func calls needed in this case?  Would this be more clean if the rhs was turned into an arrow table in a manner similar to the next test?

Reply:
No, see my comment from 82-84 on comparing arrow tables, which also applies here by comment on 102.